tests/unit_test.py

```diff
@@ -75,3 +92,148 @@ def test_default_unit_for_string_is_none():
                                sc.Unit('12.3 * m/A*kg^2/rad^3'), sc.Unit('count')))
 def test_dict_roundtrip(u):
     assert units_identical(sc.Unit.from_dict(u.to_dict()), u)
+
+
+@pytest.mark.parametrize('unit_type', (str, sc.Unit))
+def test_unit_alias_overrides_str_formatting(unit_type):
+    sc.units.aliases['clucks'] = unit_type('19.3 m*A')
```

Comment:
I wonder if going via strings (and units) here is the correct choice. Have you considered using a quantity, i.e., scalar variable? If we made this string based it may lead to tricky situations, e.g., with fractions "2/3 m" or something like that.

```python
sc.units.aliases['clucks'] = sc.scalar(19.3, 'm*A')
```

Reply:
I did not consider this. And I think it is fundamentally different, because we define units here, not quantities.

I don't see why this would be tricky. We do not add any new capability for string parsing, users can already put numbers in their units. And there is likely no need to do this dynamically. Units are known ahead of time and it is possible to carefully write strings by hand in all cases.